src/main/java/org/terasology/logic/inventory/StartingInventorySystem.java

```diff
@@ -55,121 +59,98 @@ public void initialise() {
         blockFactory = new BlockItemFactory(entityManager);
     }
 
-    @ReceiveEvent(components = {InventoryComponent.class})
+    @ReceiveEvent
     public void onStartingInventory(OnPlayerSpawnedEvent event,
                                     EntityRef entityRef,
                                     StartingInventoryComponent startingInventoryComponent) {
-        InventoryComponent inventoryComponent = entityRef.getComponent(InventoryComponent.class);
-        if (entityRef.getParentPrefab() != null) {
-            logger.info("Adding starting inventory to {}, entity has {} slots",
-                    entityRef.getParentPrefab().getName(), inventoryComponent.itemSlots.size());
-        }
+
+        InventoryComponent inventoryComponent =
+                Optional.ofNullable(entityRef.getComponent(InventoryComponent.class))
+                        .orElse(new InventoryComponent(40));
```

Comment:
Should the inventory size ("40") come from somewhere else instead of being hard-coded here?

Reply:
The number 40 is kind of magic here, that's right. It's an arbitrary number of inventory size to be used if no `InventoryComponent` is present yet. 
I could think of something like this:

```json5
{
  "StartingInventory": {
    "slots": 40,
    "items": [ ... ]
  }
}
```

`slots` could be optional, defaulting to 40 or whatever other number. Remaining question is what the behavior should be of specifying `slots` if an `InventoryComponent` is already present - should it override, or be ignored? A third option would be to ignore the starting inventory altogether with the entity does not have an inventory...

Personally, I'd vote for override, but I'm not sure about this.

Reply:
I think I'd vote for the ignore the starting inventory" way.
`InventoryComponent` not being present brings up the question _why_ it's not present.
I believe the default player prefab has an `InventoryComponent` at the moment, does it? If so, either it was deliberately removed, in which case we shouldn't add it (shouldn't we?). Or somebody changed the player prefab in the future, e.g. so that the inventory creating is done via the `StartingInventorySystem` generally, in which case it should not be ignored anymore, thus the behavior here should be adapted which would be fine IMO 🤔 

If the default player prefab currently doesn't have an `InventoryComponent`, then that's not an option, in which case I'd go for overriding, too. I'm afraid that will lead to strange behavior if something currently uses this magic number somewhere else instead of using something like `inventory.slotCount` which would be bad - so it'd probably be a good thing if we notice strange behavior which points us to such cases 😅

Reply:
Currently we have a _prefab delta_ for the player here in Inventory. It defines the "default inventory" with 40 slots like this: https://github.com/Terasology/Inventory/blob/develop/deltas/engine/prefabs/player.prefab#L3-L44

With the addition of `slots` this part of the delta prefab might be deleted and replaced by instead.

```json
{
  "StartingInventory": {
    "slots": 40
  }
}
```